### ISSUE TITLE
Syntax error when `RUF010` autofixes `f"{str({})}"`

### ISSUE THREAD

**Issue by harupy** [astral-sh/ruff#5530]
## Reproduction

```sh
> ruff --version
ruff 0.0.275

> echo 'f"{str({})}"' | ruff --select RUF010 --stdin-filename a.py --fix -

error: Autofix introduced a syntax error. Reverting all changes.

This indicates a bug in `ruff`. If you could open an issue at:

    https://github.com/astral-sh/ruff/issues/new?title=%5BAutofix%20error%5D

...quoting the contents of `a.py`, the rule codes RUF010, along with the `pyproject.toml` settings and executed command, we'd be very appreciative!

f"{str({})}"
```

**Comment by charliermarsh** [comment]
Oh boy. Did this trigger on a real project? Just curious how common it is.

**Comment by harupy** [comment]
> Did this trigger on a real project?

Yes, on https://github.com/mlflow/mlflow.


> Just curious how common it is.

I think it's uncommon.

**Comment by charliermarsh** [comment]
Haha we should still fix it definitely, was just looking for examples :)

**Comment by harupy** [comment]
The original code looked like this:

```python
"foo bar baz {}".format(str({}))
```

I first applied `UP032`:

```python
f"foo bar baz {str({})}"
```

Then, applied `RUF010` and hit the error.

**Comment by harupy** [comment]
`f"{{}!s}"` is invalid, but `f"{{}}"` is.

```python
>>> f"{{}}"
'{}'
>>> f"{{}!s}"
  File "<stdin>", line 1
SyntaxError: f-string: single '}' is not allowed
```

**Comment by charliermarsh** [comment]
Hmm yeah but they have different meanings, in the first case it's being interpreted as the literal braces whereas in the second it's an expression.

E.g. consider if the original example was: `"foo bar baz {}".format(str({"a": 1}))`. Then `f"foo bar baz {{"a": 1}}"` would be invalid syntax.
